### PR TITLE
Script for reproducing issue 1862

### PR DESCRIPTION
**Pull request by HSavynetska**
ATF Test Scripts to check #[1862](https://github.com/smartdevicelink/sdl_core/issues/1862)

This PR is **ready** for review.

### Summary
ATF test script to verify the issue.

### ATF version
[develop](https://github.com/smartdevicelink/sdl_atf/commit/6ed5c5c7a55fa05b015de3db7c04e13b91ace01a)

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: test_scripts/Defects/4_6/1862_SDL_does_not_validate_invalid_characters_in_HMI_requests.lua [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+end
+
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", common.preconditions)
+runner.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+runner.Step("RAI, PTU", common.rai_ptu)
+
+runner.Title("Test")
+for k, v in pairs (invalidCharacters, Value) do
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/4_6/1862_SDL_does_not_validate_invalid_characters_in_HMI_requests.lua]: pairs(table) function has one parameter table. Please remove Value from call pairs (invalidCharacters, Value)

File: test_scripts/Defects/4_6/1862_SDL_does_not_validate_invalid_characters_in_HMI_requests.lua [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+local common = require('test_scripts/Defects/commonDefects')
+
+--[[ Local Variables ]]
+local invalidCharacters = {
+  NewLine = "\n",
+  Tab = "\t",
+  OnlyWhitespace = " "
+}
+
+local Value = "StatusUpToDate"
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/4_6/1862_SDL_does_not_validate_invalid_characters_in_HMI_requests.lua]: please remove, this value is not used

File: test_scripts/Defects/4_6/1862_SDL_does_not_validate_invalid_characters_in_HMI_requests.lua [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", common.preconditions)
+runner.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+runner.Step("RAI, PTU", common.rai_ptu)
+
+runner.Title("Test")
+for k, v in pairs (invalidCharacters, Value) do
+  if k ~= "OnlyWhitespace" then
+   Value = "StatusUpToDate" .. v
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/4_6/1862_SDL_does_not_validate_invalid_characters_in_HMI_requests.lua]: local Value must be declared in such way:
```
for k, v in pairs (invalidCharacters) do
  local Value
  if k ~= "OnlyWhitespace" then
   Value = "StatusUpToDate" .. v
  else
    Value = v
  end
  runner.Step("GetUserFriendlyMessage_request_with_invalid_character_" .. k, GetUserFriendlyMessage, { Value })
end
```

File: test_scripts/Defects/5_0/1862_SDL_does_not_validate_invalid_characters_in_HMI_requests.lua [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+end
+
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", common.preconditions)
+runner.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+
+runner.Title("Test")
+runner.Step("GetUserFriendlyMessage valid params", testGetUserFriendlyMessageValidParams)
+runner.Step("GetUserFriendlyMessage madatory endline", testGetUserFriendlyMessageRequest, {
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/1862_SDL_does_not_validate_invalid_characters_in_HMI_requests.lua]: typo madatory -> mandatory also in 45 and 50 lines

Comment (dboltovskyi) [reply test_scripts/Defects/5_0/1862_SDL_does_not_validate_invalid_characters_in_HMI_requests.lua]: @GetmanetsIrina Done in f6c6977

**Comment by GetmanetsIrina** [comment]
PR is rebased on current develop(a7db5cc) branch.

**Comment by GetmanetsIrina** [comment]
Script is applicable for all policy flows and transport types.
Script has been moved to 6_1 folder.
Test set for script is added.

**Review by JackLivio — APPROVED** [review]

**Pull request merged**